Starfish H1 2019 Roadmap (#895)
spacetx/starfish: docs/source/roadmap.md

1. Data is easily converted into SpaceTx format from common imaging formats
2. Library that can be used to reproduce collaborator's results for single fields of view
3. Library is usable by SpaceTx for generating pipeline recipes on a single or small number of FOVs (minimum) or processing an entire experiment on individual machines (at best)
4. A way for the starfish team to run starfish at scale to process the initial set of SpaceTx datasets
5. Documentation that supports new users to learn to use starfish without our involvement
6. Starfish facilitates easy interaction with external visualization software (Napari, FIJI) to enable users to tune image processing parameters
7. Experiment-scale output formats can be loaded on local computers and are adequately expressive to enable QC workflows and specified by our users.
8. Starfish output formats enable external annotation to place the data in anatomical and cell type context and allows comparison across methods

Each deliverable above can be broken up into a set of milestones
### Library that can be used to reproduce collaborator's results for single fields of view

> 1. Obtain example data, pipelines, and results from each SpaceTx group (9/10)<sup>[1](#fn1)</sup>

Author: Added a sentence below header that is referred to here as line 41.

6. Implementation, in starfish, of the same single-FOV pipelines above (3/10)

### Library is usable by SpaceTx for generating pipeline recipes (minimum) or processing their data on individual machines (at best)
1. Library is easily installed on collaborators computers, regardless of OS (osx, linux, windows) using their choice of package manager (pip, conda)
2. Basic for-loop implementation of a pipeline runner to de-risk scale problems that result from processing multiple fields of view, then integrating results for biological analysis
3. Explicit definition/specification of a pipeline recipe
4. API for processing multiple FOVs for each SpaceTx group given data in SpaceTx format and a pipeline recipe
5. Translate each example pipeline into an example pipeline recipe
6. Visualization tooling to enable parameter selection on small datasets to tweak pipeline recipes.
7. API leverages local parallelism

### A way to run starfish at scale to process SpaceTx datasets
1. Starfish leverages parallelism to rapidly process multiple FOVs at scale (0/9)<sup>[4](#fn4)</sup>

### Documentation such that new users can learn to use it without our involvement

> 1. Public and Private API is fully typed and documented

Reviewer: I understand the need for documentation of the private API, but it does not seem to support that "new users can learn to use it without our involvement" use case.

If we want to track it in the roadmap (along with other developer-specific items), perhaps we need an additional bucket in the roadmap?

> 1. Public and Private API is fully typed and documented

Author: Ok, this makes sense to me. @ttung do you think something akin to "private API is adequately documented to support contributions from external developers" is a 0.1.0 thing, or a future goal?

> 1. Public and Private API is fully typed and documented

Reviewer 2: I suspect it is a nice-to-have for 0.1.0, but not a hard requirement.

> 1. Public and Private API is fully typed and documented

Reviewer: Awesome. If it's not a hard requirement, I vote that we drop it from the roadmap.

2. Purpose of each pipeline component is described, and guidance on parameter tuning, if necessary, is provided
3. There are examples of formatting data in SpaceTx format
4. There are examples of fully-worked pipelines
5. There are examples of the workflow for creating new pipelines
6. Public API is defined

> 7. Public API is stable

Reviewer 2: Frozen is bad.  Stable is good.

### Scientific assessment of analyzed SpaceTx data: cell type mapping
The SpaceTx consortium Working Group 5 is devoted to mapping the cells in Cell x Gene tables into the transcriptomic types/clusters delivered by Working Group 2.  _starfish_ will define standard output formats for external (non-_starfish_) calculation of these cell types and subsequent incorporation of cell type information as metadata in the Cell x Gene tables.

### Scientific assessment of analyzed SpaceTx data: methods comparison and meta analysis
The SpaceTx consortium Working Group 6 is devoted to comparing the results of each method and assessing their relative strengths and weaknesses. These are largely scientific analyses enabled by the standardization of each assay's outputs by _starfish_.  Because each assay outputs an IntensityTable, Segmentation Mask, and Expression Matrix, it will be simple to label cells or spots with anatomical information represented as named 2D polygons.

> This capability will allow straightforward creation of highly compressed datasets that will enable working Group 6 to ask questions across all methods like "how many cells/mm2 belonging to the Sst-Chodl cluster were found in layer 4 of mouse primary visual cortex?"  or "Within the inhibitory cell class, what is the abundance and distribution of Pvalb+ and VIP+ cells in human cortex?"

Author: Is this clearer? 
```
Starfish's capability to create these highly compressed output formats will enable 
working Group 6 to ask questions across all methods like... 
```

> This capability will allow straightforward creation of highly compressed datasets that will enable working Group 6 to ask questions across all methods like "how many cells/mm2 belonging to the Sst-Chodl cluster were found in layer 4 of mouse primary visual cortex?"  or "Within the inhibitory cell class, what is the abundance and distribution of Pvalb+ and VIP+ cells in human cortex?"

Reviewer 3: I think this helps, but I think it's still unclear what is meant by "highly compressed"? Is the idea that the output data are much smaller because the images have been transformed into a set of features (i.e., decoded spots, cell outlines, and expression matrices)? Or is the main point that the standardized output allows direct comparison across experiments, whereas when each group uses their own analysis package, the data needs to be wrangled first?

> This capability will allow straightforward creation of highly compressed datasets that will enable working Group 6 to ask questions across all methods like "how many cells/mm2 belonging to the Sst-Chodl cluster were found in layer 4 of mouse primary visual cortex?"  or "Within the inhibitory cell class, what is the abundance and distribution of Pvalb+ and VIP+ cells in human cortex?"

Author: It's both, good point. I've amended this section.

Finally, with a standard format to save and load Cell x Gene tables, visualization of this data will be relatively straightforward for analysts in contributor labs and Working Group 6 members.

## Key Deliverables
We believe that delivering the following features will enable the above use cases.

1. Data is easily converted into SpaceTx format from common imaging formats

> 2. Library that can be used to reproduce collaborator's results for single fields of view

Author: Good point. It is considered an important deliverable, and we will add it.

2. 2D polygon annotations of anatomical structures in physical coordinates can be saved in a standardized format and membership of each cell in annotation structures can be added as metadata to the Cell x Gene table
3. Cell x Gene table can be easily imported by computational biologists (outside of _starfish_) to determine cell type or cluster membership for each cell.  This cluster membership can include probabilistic membership to multiple clusters and is added to the Cell x Gene table as metadata

### A tool to convert image data into SpaceTx Format
1. CLI and API (Java) can parse [supported file formats](https://docs.openmicroscopy.org/bio-formats/6.0.0-m3/supported-formats.html) into SpaceTx-formatted datasets<sup>[5](#fn5)</sup>
2. Installation of the tool can be accomplished by downloading an archive with a launcher script or by using pre-built docker images
3. Users and/or data wranglers can point the tool at one of the [stated files](https://docs.openmicroscopy.org/bio-formats/6.0.0-m3/formats/dataset-table.html) for each FOV to generate 2D TIFFs as well as the necessary SpaceTX JSON files

> 4. In the case that multiple “series” are present in a single dataset, the user must specify which series to use

Author: @joshmoore

> 4. In the case that multiple “series” are present in a single dataset, the user must specify which series to use

Reviewer 4: Commit pushed directly to your branch.

### A Modular Library of Pipeline Components

#### Registration
We lack solutions for registration that fulfill requirements for ISS (the Harris version), smFISH, MERFISH. Additionally, expansion microscopy requires non-affine thin-plate spline registration. Registration is out of scope for the timeline expected by SpaceTx to have all the data processed. As such, we expect all contributors to provide us with either pre-registered data, or the registration pipeline component implementations they require. We will need to develop solutions or compromises for the MExFISH and ISS (Harris) methods.

#### Documentation
_starfish_'s API documentation is minimal. We need to add basic documentation that describes how to use each component, and how to fit parameters. This should be adequate for a skilled computational user (Brian Long, Kenneth Harris) to pick up _starfish_ and use it to make a pipeline. _starfish_ should also clearly document how a developer can contribute code they need for their analyses to the project. We will need to source feedback from users on what parts of the documentation provoke confusion.

#### Image re-scaling & normalization

> Several approaches normalize images to overcome different channel intensities or round biases. We should implement a pipeline component to support this that samples from images within or across fields of view. Determine the minimum necessary sampling level to achieve needed normalization.

Reviewer 3: Can you clarify what you mean by "minimum necessary sampling level"?

Author: Is the following clearer? 

```diff
- Determine the minimum necessary sampling level to achieve needed 
- normalization.
+ Determine the minimum amount of data to sample from images across 
+ the experiment to equalize their intensities.

Reviewer 3: Looks good to me!

#### Affine Registration
For the SpaceTx pilot project, it is adequate to apply pre-computed transformations or leverage already-registered data. Later, when we want to support groups to use _starfish_ for their own research, it will be important to implement affine registration, as otherwise _starfish_ serves only part of their use cases, and they are incentivized to continue running their existing pipelines.

#### Non-Affine Registration
Non-Affine registration approaches are very specific to the types of deformations that the tissue suffers, and have variable forms. We should investigate defining some kind of API for arbitrary transformations that could allow users to add approaches that they require, but we won't do the work to implement these as they don't have broad applicability.

#### Fixing Chromatic Aberration

> Chromatic aberrations, including differences in illumination, cross-talk between channels, and channel drift tend to be specific to the assay and microscope, making creation of a general solution complicated. Additionally, these problems are often fixed early in pipelines so it is easy to extricate this step from the pipeline. For these reasons, we will ask the groups to generate data that has already been corrected for release 0.1.0.

Reviewer 3: While it is true that these corrections can be more easily performed outside of starfish because they are early in the workflow, I think they really should be done in starfish so they are logged. These types of corrections can have a large effect on the final result (e.g., pixel wise decoding using a euclidian distance to assign targets assumes certain relative intensities between channels/rounds), so I think it is important that they are logged with the rest of the processing steps to improve reproducibility (between experiments, not re-running an analysis) and comparison between datasets.

Also, I think that (technically) cross-talk and a lot of the differences in measured intensity are not chromatic aberrations :P

Author: > Also, I think that (technically) cross-talk and a lot of the differences in measured intensity are not chromatic aberrations :P

What would you call them instead? 

> While it is true that these corrections can be more easily performed outside of starfish because they are early in the workflow, I think they really should be done in starfish so they are logged. These types of corrections can have a large effect on the final result (e.g., pixel wise decoding using a euclidian distance to assign targets assumes certain relative intensities between channels/rounds), so I think it is important that they are logged with the rest of the processing steps to improve reproducibility (between experiments, not re-running an analysis) and comparison between datasets.

Completely agree 👍. We intend to include some solution(s) for these problems (and therefore add them to provenance logging), but we don't have enough computational biology hours in the coming quarter, and we won't have time to add these capabilities prior to 0.1.0. 

That said, if you wanted to _contribute_ such solutions needed for _your system_, that would be very valuable and a model for future groups :-).

Reviewer 3: >> Also, I think that (technically) cross-talk and a lot of the differences in measured intensity are not chromatic aberrations :P
> 
> What would you call them instead?

Chromatic aberration arises from changes in refractive index as a function of wavelength (i.e., dispersion). Cross-talk between different channels is due to the microscope filters not rejecting fluorescence from other dyes (may not be possible due to overlap in spectra). The difference in intensities across channels is usually due to differences in illumination intensity, properties of the dyes, and the sensitivity of the detector. All of that being said, I can't think of a catch all term at the moment :P

After re-reading, I am confused about what this section refers to. Is this section proposing ways to automatically determine the transformations required to make the corrections or is it proposing to apply corrections determined/defined the by user? If it's the latter, I think there is a lot of overlap with the "Image re-scaling & normalization" section.

Reviewer 3: > That said, if you wanted to contribute such solutions needed for your system, that would be very valuable and a model for future groups :-).

I am definitely interested. We need to do some cross-talk correction in upcoming experiments, so I can work on at least that part soon.

Author: > Is this section proposing ways to automatically determine the transformations required to make the corrections or is it proposing to apply corrections determined/defined the by user? If it's the latter, I think there is a lot of overlap with the "Image re-scaling & normalization" section.

Like other stuff in this section, I have less experience with these image processing. It's "one or the other, depending on how hard they turn out to be". I think this is worth bringing up at the sync, as there's complexity here I'd like to unpack through conversation.

Reviewer 3: Sounds good! I look forward to discussing at the sync.

    1. ISS Breast cancer. 16 FOVs with shape (4, 4, 1, 1024, 1024),
    2. MERFISH U2-OS cells. 400 FOVs with shape (8, 2, 1, 2048, 2048),
    3. osmFISH Visual Cortex. ~100s FOVs with shape (13, 3, 45, 2048, 2048)
2. Outputs from Fields of view can be combined into a single output format
3. Output formats can be loaded and are performant
4. Processed data can be used to recapitulate results

### _starfish_ output formats enable external annotation to place the data in anatomical and cell type context and allow comparison of across methods
1. Processed Experiments (all FOVs for a given dataset) can produce Cell x Gene tables with each cell in the physical coordinates of the sample
2. 2D polygon annotations of anatomical structures in physical coordinates can be saved in a standardized format and membership of each cell in annotation structures can be added as metadata to the Cell x Gene table
3. Cell x Gene table can be easily imported by computational biologists (outside of _starfish_) to determine cell type or cluster membership for each cell.  This cluster membership can include probabilistic membership to multiple clusters and is added to the Cell x Gene table as metadata

> ### A tool to convert image data into SpaceTx Format

Author: @joshmoore leaving you a TODO to translate OME terminology to SpaceTx when you have a translation you're happy with.

# _starfish_ Roadmap
This document describes the features and timeline for current work on the _starfish_ package.

> The purpose of this document is to describe the deliverables that we intend to deliver by June 2019. To accomplish this, we first outline our primary [use cases](#use-cases). With these to guide us, we then define milestones for releases that support these needs [**what we need to build**](#what-we-need-to-build:-starfish-0.1.0), and candidate features which we may consider building in the future [**what we can build later**](candidates-features-for-0.2.0+).

Reviewer 2: Potential point of confusion: What is the significance of June 2019?  If this is obvious to everyone who might read this document, we are good.  Otherwise, it's worth expanding.

## Use cases
_starfish_ is being developed to support the data processing and benchmarking needs of the SpaceTx consortium and the Chan Zuckerberg Biohub. We have identified a set of use cases from these groups that _starfish_ will support. Feedback from these users will set the course for future work on _starfish_, beyond the milestones identified here.

### Local processing of data

> Some image-based transcriptomics asays have moderate processing requirements. Users of these assays would like to be able to use _starfish_ to process multiple fields of view of data on their local machines to enable further parameter tuning and spot-checking of results. To support this, we will define a Python API for running _starfish_ on local machines that processes experiments of arbitrary numbers of fields of view, logs, stores data provenance, is multi-processing enabled. Users will also be able to run _starfish_ pipelines through a command-line interface (CLI).

Reviewer 2: ```suggestion
Some image-based transcriptomics assays have moderate processing requirements. Users of these assays would like to be able to use _starfish_ to process multiple fields of view of data on their local machines to enable further parameter tuning and spot-checking of results. To support this, we will define a Python API for running _starfish_ on local machines that processes experiments of arbitrary numbers of fields of view, logs, stores data provenance, is multi-processing enabled. Users will also be able to run _starfish_ pipelines through a command-line interface (CLI).
```

> Some image-based transcriptomics asays have moderate processing requirements. Users of these assays would like to be able to use _starfish_ to process multiple fields of view of data on their local machines to enable further parameter tuning and spot-checking of results. To support this, we will define a Python API for running _starfish_ on local machines that processes experiments of arbitrary numbers of fields of view, logs, stores data provenance, is multi-processing enabled. Users will also be able to run _starfish_ pipelines through a command-line interface (CLI).

Reviewer 2: > To support this, we will define a Python API for running _starfish_ on local machines that processes experiments of arbitrary numbers of fields of view, logs, stores data provenance, is multi-processing enabled.

this sentence is confusing.

### Parameter selection on single fields of view

> Image-based transcriptomics workflows are highly dependent upon the tissue, organism, and probes being assayed. Researchers typically select image processing parameters by analyzing and optimizing parameter selection in a single field of view. In support of this use case, _starfish_ must enable users to process data for any SpaceTx assay on the local computer of their choosing and interact with visualization tools necessary to evaluate parameter decisions. It must support this functionality for users that are not proficient in Python.

Reviewer 2: terminology question: why is it sometimes "image-based transcriptomics assays" and sometimes "image-based transcriptomics workflows"?

Author: This is a good question. "workflow" is overloaded. I am using it here to refer to a file that records the steps of computational analysis in a pipeline. In contrast, the assays are the data acquisition steps.

### Scalable processing of SpaceTx assays

> It may not always be possible or time efficient to process an entire experiment on a local machine, therefore _starfish_ must be able to process a complete experiment consisting of hundreds of fields of view that are comprised of up to 10s of terabytes of 2-d TIFF images. For the SpaceTx consortium project, the _starfish_ team agreed to process the data for each of the data contributors, so the _starfish_ team will define a "Scalable Pipeline Runner" that will run _starfish_ pipelines on a parallel computing infrastructure of our choice. We will expose tooling to explore the output formats on a personal computer of our contributor's choosing.

Reviewer 2: ```suggestion
It may not always be possible or time efficient to process an entire experiment on a local machine. Therefore, _starfish_ must be able to process a complete experiment consisting of hundreds of fields of view that are comprised of up to 10s of terabytes of 2-d images. For the SpaceTx consortium project, the _starfish_ team agreed to process the data for each of the data contributors, so the _starfish_ team will define a "Scalable Pipeline Runner" that will run _starfish_ pipelines on a parallel computing infrastructure of our choice. We will expose tooling to explore the output formats on a personal computer of our contributors' choosing.
```

### A General Format and Object Model for Image-based Transcriptomics
_starfish_ defines a general set of data formats (and paired python object implementations) that (1) enable efficient storage and access for a related set of images broken up into single fields of view<sup>[6](#fn6)</sup>, (2) enable storage and utilization of a codebook, which defines how sets of images are combined to decode spot patterns into biological targets, and (3) defines a pipeline recipe which can be interpreted by _starfish_'s pipeline runner to process an experiment.

Currently, starfish needs images in SpaceTx Format must either be pre-aligned, such that the x-y location of each tile in a Field of View is the same. Image data that does not adhere to this requirement will not be processable with _starfish_. See [Candidate features for future releases](#candidate-features-for-future-releases) for more discussion of future directions here.

### A Modular Library of Pipeline Components

> _starfish_ enables the construction of arbitrary image-based transcriptomics pipelines by implementing the union of algorithms used by SpaceTx labs while combining similar approaches into a simplified subset. These algorithms can be linearly combined to process individual fields of view, requiring no more than 16GB ram for any individual pipeline component. The set of algorithms must include:

Reviewer 2: Is this 16GB reserved for starfish?  If we are trying to fit within some machine type on GS/EC2, we need to take into account what the OS and other services demand.

Author: Made this more ambiguous. Should be "runs on a laptop".

#### Image Adjustment
_starfish_ defines pipeline components that enable simple adjustment of image data, such as slicing (finding focal plane, cropping edge effects) and projection over arbitrary axes.

#### Filtering
_starfish_ defines the set of filters used in SpaceTx labs, including approaches to remove background, enhance spots, deconvolve the specified point spread function of the imaging system, and normalize image intensities.

#### Spot Detection

> _starfish_ implements spot detectors via a variety of spot detection algorithms (eg., Gaussian/Laplacian of Gaussian/Hessian of Gaussian blob detection, local max peak finding, etc.) These detectors can function independently on individual channels and rounds or be used to measure intensity values of spots (as detected in a single auxiliary image) across all other primary images across channels and rounds. _starfish_ also supports pixel based decoding algorithms (e.g., algorithms that detect spots after pixels are decoded to targets) within the same API.

Reviewer 2: ```suggestion
_starfish_ implements spot detectors via a variety of spot detection algorithms (e.g., Gaussian/Laplacian of Gaussian/Hessian of Gaussian blob detection, local max peak finding, etc.) These detectors can function independently on individual channels and rounds or be used to measure intensity values of spots (as detected in a single auxiliary image) across all other primary images across channels and rounds. _starfish_ also supports pixel based decoding algorithms (e.g., algorithms that detect spots after pixels are decoded to targets) within the same API.
```

#### Decoding

> _starfish_ enables decoding of spots or pixels detected across rounds and channels into targets. These decoding results can be filtered post-hoc, using spot intensity, spot size, and spot distance-to-target metrics. For the latter metric, the user may supply their desired distance metric.

Reviewer 2: ```suggestion
_starfish_ enables decoding of spots or pixels detected across rounds and channels into targets. These decoding results can be filtered post-hoc, using spot intensity, spot size, and spot distance-to-target metrics. For the last metric, the user may supply their desired distance metric.
```

#### Segmentation
This is an area of active research. However, four SpaceTx groups use seeded watershed segmentation, which has a concrete implementation. _starfish_ implements this method and will include a vignette on how to tune parameters so it functions across assays. In addition, we will elicit submissions from Kenneth Harris and Peter Kharchenko for point-cloud based segmentation, and encourage the community to work on this problem by defining a segmentation format and API to enable plugin-based interaction with community-developed algorithms.

#### Image Registration

> _starfish_ implements the ability to apply a pre-computed full affine registration including translation, rotation, and scale to enable alignment across z-layers, rounds, and channels. Translation and rotation can also provide a solution for some types of chromatic aberration.

Reviewer 2: wait it does?

or this is something we plan to do?  if the latter, should be future tense.

### Worked examples that process each assay

> We need to demonstrate that _starfish_ is capable of processing each SpaceTx assay. To do this, we will create pipelines that process single fields of view for each group. These pipelines will take as input datasets that have already been registered and for which chromatic aberrations have been corrected, and will output spot calls. To validate that our results match SpaceTx expectations, we will calculate the copy number correlation between our results and the results of their own pipelines, as well as overlay our spot calls on the image data.

Reviewer 2: ```suggestion
We need to demonstrate that _starfish_ is capable of processing each SpaceTx assay. To do this, we will create pipelines that process single fields of view for each group. These pipelines will take as input datasets that have already been registered and for which chromatic aberrations have been corrected, and will output detected spots. To validate that our results match SpaceTx expectations, we will calculate the copy number correlation between our results and the results of their own pipelines, as well as overlay our spot calls on the image data.
```

#### _starfish_ must enable processing of fields of view by round
Sequential smFISH assays that capture volumetric images produce fields of view that are too large for personal computers<sup>[7](#fn7)</sup>. However, if we are able to break up the processing of this data by round, we could hit our scale targets. Second, multiplexed assays (e.g. MERFISH, SeqFISH, ISS) require all rounds and channels to be loaded into memory to decode. In both cases, we should derisk decoding of these assays by understanding the maximum ImageStack size we can support before we need to implement solutions to manage memory consumption.

### A Modular Library of Pipeline Components

#### Documentation
We need to add basic documentation that describes how to use each component, and how to fit parameters. This should be adequate for a skilled computational user to pick up _starfish_ and use it to make a pipeline. _starfish_ should also clearly document how a developer can contribute code they need for their analyses to the project. We will need to source feedback from users on what parts of the documentation provoke confusion.

> #### Image re-scaling & normalization

Reviewer 2: Is reslicing existing data sets and persisting them onto disk something we need to do?  We now have a 3000px cap on the tiles, which may or may not be an issue.

Author: I think this is best answered by looking at the sizes of the fields of view we are being given by each contributor and the requirement that they need to run on a laptop. So long as the laptop requirement is fulfilled and the library is fast, the implementation/execution is up to you.

5. The starting index of the field of view may be changed from the default of zero
6. If a codebook exists at the time of creation, it can be passed to the tool for inclusion in the JSON. Otherwise, a dummy codebook will be created

## Candidate features for future releases

There are many other features that are good candidates for future releases. We discuss these features, and why we believe they can be built at a later point.

### Hardware- and Acquisition-related corrections

We lack general purpose solutions for registration that fulfill requirements for ISS (the Harris version), smFISH, & MERFISH. Additionally, expansion microscopy requires non-affine thin-plate spline registration. Registration is out of scope for the timeline expected by SpaceTx to have all the data processed.

In order to benchmark datasets, all contributors will need to provide pre-processed images before uploading. This will help us focus on _starfish_'s key value proposition and meet our goal of comparing the results from all the methods using standard file formats, and scalable, reproducible image processing workflows.

> A key challenge in developing generalized solutions to registration and other image processing corrections is that the necessary corrections are highly specific to the signal acquisition challenges each data generator faces. Furthermore, proper image pre-processing is critical to the success of the downstream pipeline components that we're building in the roadmap defined above. Because data generators understand their hardware, chemistry, and samples much better than we do, for now, they are best able to apply their solutions before the data goes into _starfish_ pipelines.

Reviewer 3: @ambrosejcarr I think we can have the field flatness/image normalization (`ElementWiseMult()`) and the `LinearUnmixing` components landed by the 0.1.0 date (April 2019).
